### PR TITLE
[E-Ivan < T393] Implement from_values on map

### PR DESCRIPTION
**Pull request by imilinovic**
### Description
Implement map.from_values

### Pull request type

- [ ] Bugfix
- [X] Algorithm/Module
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

### Related issues

Delete if this PR doesn't resolve any issues. Link the issue if it does.

######################################

#### Module/Algorithm
- [ ] Core algorithm/module implementation
- [ ] [Query module](https://memgraph.com/docs/memgraph/reference-guide/query-modules) implementation
- [ ] Unit tests
- [ ] End-to-end tests
- [ ] Code documentation
- [ ] README short description
- [ ] Documentation on [memgraph/docs](https://github.com/memgraph/docs)
- [ ] Update GQLALchemy signatures in [query builder](https://github.com/memgraph/gqlalchemy/blob/main/gqlalchemy/graph_algorithms/query_builder.py  ) using [query module signature generator](https://github.com/memgraph/gqlalchemy/blob/main/scripts/query_module_signature_generator.py)

######################################

### CODE REVIEW

File: cpp/map_module/algorithm/map.cpp [hunk 1]
```diff
@@ -1 +1,36 @@
...
+      throw mgp::ValueException("List needs to have an even number of elements");
+    }
+
+    auto iterator = values.begin();
+    while(iterator != values.end()) {
+      auto &key = *iterator;
+      if (!key.IsString()) {
+        throw mgp::ValueException("Keys need to be strings");
+      }
+      ++ iterator;
```
Comment (antoniofilipovic) [inline cpp/map_module/algorithm/map.cpp:24]: `++iterator`

File: cpp/map_module/algorithm/map.cpp [hunk 2]
```diff
@@ -1 +1,36 @@
...
+      map.Update(key.ValueString(), *iterator);
+      ++ iterator;
```
Comment (antoniofilipovic) [inline cpp/map_module/algorithm/map.cpp:26]:
```suggestion
      ++iterator;
```

File: cpp/map_module/algorithm/map.cpp (hunk 1 above)
Comment (antoniofilipovic) [inline cpp/map_module/algorithm/map.cpp:24]:
```suggestion
      ++iterator;
```

**Review by antoniofilipovic — CHANGES_REQUESTED** [review]
Small changes, otherwise looks good

File: cpp/map_module/algorithm/map.cpp [hunk 3]
```diff
@@ -1 +1,86 @@
 #include "map.hpp"
+#include <string>
+#include "mg_utils.hpp"
+#include "mgp.hpp"
+
+namespace {
+std::string value_to_string(const mgp::Value &value) {
```
Comment (imilinovic) [inline cpp/map_module/algorithm/map.cpp:7]: This function supports key being any type that isn't null, list, map or path. There wasn't a simple way to implement it (except null which can be ignored) but I don't think it's very important at the moment.
This function could perhaps be added inside mgp::Value

File: cpp/map_module/algorithm/map.cpp [hunk 4]
```diff
@@ -1 +1,86 @@
...
+    case mgp::Type::Int:
+      return std::to_string(value.ValueInt());
+    case mgp::Type::Double:
+      return std::to_string(value.ValueDouble());
+    case mgp::Type::String:
+      return std::string(value.ValueString());
+    case mgp::Type::List:
+      return "";
+    case mgp::Type::Map:
+      return "";
```
Comment (antoniofilipovic) [inline cpp/map_module/algorithm/map.cpp:24]: Can we support that via recursive calls? I think it makes sense to add that as part of mgp::Value and maybe overload operator << for mgp::Value directly. What do you think?

**Review by antoniofilipovic — APPROVED** [review]

**Pull request merged**